### PR TITLE
added more just commands

### PR DESCRIPTION
**Pull request by boredsquirrel**
- cups disable
- rebasing between userns and no-userns variant, easy due to the good naming convention and json output of `rpm-ostree status`
- bluetooth on/off
- setting a generic hostname
- switching between flathub-verified and flathub-all

also I chained all commands together with success messages if succeeded.

### CODE REVIEW

**Comment by boredsquirrel** [comment]
the flathub switch could be upstream, but only secureblue restricts flathub by default.

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 1]
```diff
@@ -2,14 +2,85 @@
...
+    rpm-ostree kargs --append="init_on_alloc=1" --append="init_on_free=1" --append="slab_nomerge" --append="page_alloc.shuffle=1" --append="randomize_kstack_offset=on" --append="vsyscall=none" --append="debugfs=off" --append="lockdown=confidentiality" --append="random.trust_cpu=off" --append="random.trust_bootloader=off" --append="intel_iommu=on" --append="amd_iommu=on" --append="efi=disable_early_pci_dma" --append="iommu.passthrough=0" --append="iommu.strict=1" --append="nvme_core.default_ps_max_latency_us=0" --append="mitigations=auto,nosmt" &&\
+    echo "Hardening kernel arguments applied."
 
 harden-flatpak:
-    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so
+    flatpak override --user --filesystem=host-os:ro --env=LD_PRELOAD=/var/run/host/usr/lib64/libhardened_malloc.so &&\
+    echo "Flatpaks are now using harened_halloc."
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:10]: please remove these unnecessary changes. all echos and all &&\ are unnecessary and not how just commands work

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 2]
```diff
@@ -2,14 +2,85 @@
...
 
 enable-cups:
-    firewall-cmd --add-service=cups --permanent 
-    firewall-cmd --reload 
-    systemctl unmask cups
-    systemctl enable cups
-    systemctl start cups
+    firewall-cmd --add-service=cups --permanent &&\
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:13]: revert all these changes

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 3]
```diff
@@ -2,14 +2,85 @@
...
+    firewall-cmd --reload &&\
+    systemctl unmask cups &&\
+    systemctl enable --now cups &&\
+    echo "CUPS enabled."
+
+disable-cups:
+    firewall-cmd --remove-service=cups --permanent &&\
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:20]: cups is disabled by default so I don't see the purpose of this just command

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 4]
```diff
@@ -2,14 +2,85 @@
...
+    firewall-cmd --reload &&\
+    systemctl stop cups &&\
+    systemctl disable cups &&\
+    systemctl mask cups &&\
+    echo "CUPS disabled."
+
+userns-on
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:27]: maintaining this will be a nightmare and it is significantly simpler for the user to just add "-userns" or remove it from their image name. please remove it.

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 5]
```diff
@@ -2,14 +2,85 @@
...
+
+    else
+        echo "The check has failed, please report that error."
+        rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"'
+        xdg-open "https://github.com/secureblue/secureblue/issues/new"
+    fi
+
+userns-off
+    # test it userns are disabled
+    if grep -q "kernel.unprivileged_userns_clone = 0" "/usr/etc/sysctl.d/hardening.conf"; then
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:49]: same thing here

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 6]
```diff
@@ -2,14 +2,85 @@
...
+        rpm-ostree rebase $nouserns_version
+        echo "Reboot for the changes to apply."
+
+    else
+        echo "The check has failed, please report that error."
+        rpm-ostree status --json | jq -r '.deployments[0]."container-image-reference"'
+        xdg-open "https://github.com/secureblue/secureblue/issues/new"
+    fi
+
+bluetooth-on
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:65]: this one is a good idea but instead please disable bluetooth in the image for all images, and then add the just command "enable-bluetooth"

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 7]
```diff
@@ -2,14 +2,85 @@
...
+    echo "Bluetooth enabled."
+
+bluetooth-off
+    systemctl stop bluetooth.service &&\
+    systmctl disable bluetooth.service &&\
+    systemctl mask bluetooth.service &&\
+    echo "Bluetooth disabled."
+
+flathub-all
+    flatpak remote-modify --subset=all flathub &&\
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:77]: aliasing these is overkill.

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 8]
```diff
@@ -2,14 +2,85 @@
...
+    echo "The Flathub Repository now shows all available Apps."
+
+flathub-verified
+    flatpak remote-modify --subset=verified flathub &&\
+    echo "The Flathub Repository now shows only verified Apps."
+
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just:83]: fedora is already generic, this is unnecessary. please remove it.

**Comment by RoyalOughtness** [comment]
As general feedback, if you want to avoid spending time on changes that may end up being rejected, you should open issues and assign them to yourself indicating you intend to work on them. That way if the changes are going to be rejected, they will be rejected before you spend time on them.

**Review by RoyalOughtness — CHANGES_REQUESTED** [review]

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 1 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:10]: Why? Isnt just just going through each line, or does it detect if there is a failure? It is important to have a confirmation after running a command, as somewhere in the middle could have been an error.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 3 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:20]: The default is disabled, but users should be always able to opt in and opt out of it. Once in, no way back? That makes no sense, and it is also not a problem to keep it at all. It should be easy to temporarily enable it, and disable it again.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 5 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:49]: it simply searches for that string, which is a general argument that doesnt change. The file where it is does not change too. 

The naming convention of the images also does not change, if it would then there would need to be a small fix, but thats not much.

I dont see how this would be "a nightmare to maintain", the only problem would be if the images would suddenly get different names, which seems unlikely.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 6 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:65]: same thing as with cups, users have to easily be able to disable it again when not using it.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 7 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:77]: agree, also there is a bug in the flatpak modify, subset=all does not exist yet and it seems that subset command does not even exist. Which is a bigger problem, I will open an issue about this.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 8 above)
Comment (boredsquirrel) [reply config/files/usr/share/ublue-os/just/60-custom.just:83]: this is the manual opt-in when users choose to use only verified apps again, e.g. when their needed apps got verified. Also this is the command that is supposed to work.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 1 above)
Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just:10]: Please have a look at how official ublue spins write just commands.

https://github.com/ublue-os/bazzite/blob/f88ff6abf15771d7fe57677ee70addcc4d688bf5/system_files/deck/shared/usr/share/ublue-os/just/60-custom.just#L4

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 3 above)
Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just:20]: The user can already use systemctl to do this. We don't need to alias things that already have simple and straightforward user experiences like systemctl.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 5 above)
Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just:49]: You are writing over 30 lines of justfiles to do something that the user can do with a one line command. It makes no sense.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 6 above)
Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just:65]: They already can with systemctl.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 8 above)
Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just:83]: This comment was on the hostnamectl just command. But on the flathub commands, they are also not needed since the user can just add a second flathub remote.

**Comment by RoyalOughtness** [comment]
Please open a new PR that disables bluetooth by default in all images. That's the only change here with significant value-add.
